src/main/java/seedu/multitasky/logic/parser/FindCommandParser.java

```diff
@@ -24,15 +27,70 @@ public FindCommand parse(String args) throws ParseException {
         String trimmedArgs = args.trim();
 
         if (args.trim().isEmpty()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                                                   FindCommand.MESSAGE_USAGE));
+        }
+        ArgumentMultimap argumentMultimap = ArgumentTokenizer.tokenize(trimmedArgs,
+                                                                ParserUtil.toPrefixArray(FindCommand.VALID_PREFIXES));
+        ArrayList<String> prefixesPresent = argumentMultimap.getPresentPrefixes(FindCommand.VALID_PREFIXES);
+        if (!hasValidPrefixCombination(prefixesPresent)) {
+            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
                                                    FindCommand.MESSAGE_USAGE));
         }
+        Set<String> keywords;
+        Optional<String> optKeywords = argumentMultimap.getPreamble();
+        if (optKeywords.isPresent()) {
+            keywords = new HashSet<>(Arrays.asList(optKeywords.get().split("\\s+")));
+        } else {
+            keywords = new HashSet<>();
+        }
+        Calendar startDate = getStartDate(argumentMultimap);
+        Calendar endDate = getEndDate(argumentMultimap);
+
+        return new FindCommand(keywords, startDate, endDate, prefixesPresent);
+    }
+
+    private Calendar getStartDate(ArgumentMultimap argumentMultimap) {
+        ArrayList<String> dateArgs = (ArrayList<String>) argumentMultimap.getAllValues(CliSyntax.PREFIX_FROM);
```

Comment:
edit: after some testing i found out `mistyped prseidnet day president day` works for pretty time lol, who knew.
but then `add things from 20 july from 31 dec` will result in `july 31` instead of `31 dec` because of the way they read month first, so i still think sticking with only last flag is a better algorithm